From 4ef0e0124f73b3bc24b2f2a772d83176aaa9fb3f Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anujx.mittal@intel.com>
Date: Tue, 8 Sep 2015 11:53:36 +0800
Subject: [PATCH] qat: replace strict_strtoull with kstrtoull

Upstream-Status: Backport

strict_strtoull() has been removed starting kernel v3.18.
Hence, replace the usage with kstrtoull().

Signed-off-by: Anuj Mittal <anujx.mittal@intel.com>
---
 .../uclo/include/linux/uclo_platform.h             |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/quickassist/utilities/downloader/Target_CoreLibs/uclo/include/linux/uclo_platform.h b/quickassist/utilities/downloader/Target_CoreLibs/uclo/include/linux/uclo_platform.h
index a4bf903..ee3e2d8 100755
--- a/quickassist/utilities/downloader/Target_CoreLibs/uclo/include/linux/uclo_platform.h
+++ b/quickassist/utilities/downloader/Target_CoreLibs/uclo/include/linux/uclo_platform.h
@@ -116,6 +116,9 @@
 /* 32-bit/64-bit 2.6.29 kernel does not export simple_strtoll */
 #define STR_TO_32(str, base, num, endPtr) {*(num) = (int)simple_strtol((str), &(endPtr), (base));}
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,18,0)
+#define STR_TO_64(str, base, num, endPtr) {endPtr=NULL; if (kstrtoull((str), (base), (num))) printk("Error strtoull convert %s\n", str); }
+#else
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
 #define STR_TO_64(str, base, num, endPtr) {endPtr=NULL; if (strict_strtoull((str), (base), (num))) printk("Error strtoull convert %s\n", str); }
 #else 
@@ -133,6 +136,7 @@
        } while(0)
 #endif
 #endif
+#endif
 
 int aToi(const char *pStr);
 
-- 
1.7.9.5

